src/GraphQL.Tests/Types/AutoRegisteringInterfaceGraphTypeTests.cs

```diff
@@ -294,7 +295,8 @@ public async Task Argument_ResolverTests_WithNonNullString(string fieldName, str
         using var provider = serviceCollection.BuildServiceProvider();
         context.RequestServices = provider;
         fieldType.Resolver.ShouldNotBeNull();
-        (await fieldType.Resolver!.ResolveAsync(context).ConfigureAwait(false)).ShouldBe(expected);
+        var ex = await Should.ThrowAsync<InvalidOperationException>(async () => await fieldType.Resolver!.ResolveAsync(context).ConfigureAwait(false)).ConfigureAwait(false);
```

Comment:
```suggestion
        var ex = await Should.ThrowAsync<InvalidOperationException>(fieldType.Resolver!.ResolveAsync(context)).ConfigureAwait(false);
```

Reply:
Does not compile -- ThrowAsync expects Task and ResolveAsync returns ValueTask

Reply:
Also, the suggested code would execute outside of the ThrowAsync, so any exceptions thrown before getting wrapped in a Task would not be caught by ThrowAsync.

Reply:
Understood. https://github.com/shouldly/shouldly/issues/753

Reply:
They advice to use `Func<Task>`.